core/src/test/scala/unit/kafka/log/LogCleanerIntegrationTest.scala

```diff
@@ -185,4 +185,30 @@ class LogCleanerIntegrationTest extends AbstractLogCleanerIntegrationTest {
       (key, curValue)
     }
   }
+
+  @Test
+  def testDeadThreadCountMetricOnUnexpectedError(): Unit = {
+    cleaner = makeCleaner(partitions = topicPartitions, maxMessageSize = 100000, backOffMs = 100)
+    cleaner.startup()
+    assertEquals(0, cleaner.deadThreadCount)
+    // we simulate the unexpected error with an interrupt
+    cleaner.cleaners.foreach(_.interrupt())
+    // wait until interruption is propagated to all the threads
+    TestUtils.waitUntilTrue(
+      () => cleaner.cleaners.foldLeft(true)((result, thread) => {
+        thread.isThreadFailed && result
+      }), "Threads didn't terminate unexpectedly"
+    )
+    assertEquals(cleaner.cleaners.size, cleaner.deadThreadCount)
+  }
+
+  @Test
+  def testDeadThreadCountMetricOnShutdown(): Unit = {
+    cleaner = makeCleaner(partitions = topicPartitions, maxMessageSize = 100000, backOffMs = 100)
+    cleaner.startup()
+    assertEquals(0, cleaner.deadThreadCount)
+    cleaner.cleaners.foreach(_.shutdown())
+    // intentionally shutting down threads shouldn't wobble the metric
+    assertEquals(0, cleaner.deadThreadCount)
```

Comment:
I'm wondering how likely this test is to catch the bug we had earlier in our discussion. If it isn't very likely, I thought it might be worth asserting in a loop for a bit. But then again, that would slow down the tests so I'm wondering whether we should have this test at all

Reply:
Yea I tend to agree, I'll remove this.